### PR TITLE
#4 Adapt getOrder method

### PR DESCRIPTION
**Pull request by 3rdvision**
| - | - |
| --- | --- |
| Issue | https://github.com/ripe-tech/epir-core/issues/4 |
| Dependencies | https://github.com/hivesolutions/yonius/pull/18 |
| Decisions | • Adapt `method` method instead of mock method. <br> • Adapt example to call `method`. <br> • Add pull request and bug report template.  |

### CODE REVIEW

File: js/order.js [hunk 1]
```diff
@@ -1,8 +1,13 @@
 export const OrderAPI = superclass =>
     class extends superclass {
-        async getOrder(number) {
+        async getOrder(number, options = {}) {
```
Comment (gcandal) [inline js/order.js:3]: This should be a dumb API client, meaning that you just `this.get(url, options)` without changing them. Clients of the API will have to build the options to contain a `X-Secret-Key` header by themselves.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+| - | - |
```
Comment (gcandal) [inline .github/PULL_REQUEST_TEMPLATE.md]: Sorry to be a PITA, but this must go in a different PR.

File: js/base.js [hunk 3]
```diff
@@ -24,6 +24,7 @@ export class API extends mix(BaseAPI).with(OrderAPI) {
         delete options.kwargs.auth;
         if (auth) {
             options.headers.Authorization = `Bearer ${this.token}`;
+            options.headers["x-secret-key"] = this.token;
```
Comment (gcandal) [inline js/base.js]: Fix the casing

**Review by gcandal — CHANGES_REQUESTED** [review]

**Review by gcandal — APPROVED** [review]

File: examples/simple.js [hunk 4]
```diff
@@ -1,10 +1,14 @@
 const epirProcessor = require("..");
 
+const baseUrl = "http://epir-processor-now.platforme.com/api/v1/";
+const token = process.env.RIPE_CORE_TOKEN;
+const orderNumber = 5555;
```
Comment (joamag) [inline examples/simple.js]: this is not ok

Comment (joamag) [reply examples/simple.js]: let's use an env variable

**Review by joamag — CHANGES_REQUESTED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**